crates/proof-of-sql/src/proof_primitive/dynamic_dory_and_hyrax_common_utils/standard_basis_helper.rs

```diff
@@ -97,33 +138,35 @@ fn build_partial_second_half_standard_basis_vecs(
     }
 }
 
-#[allow(dead_code)]
-pub(super) fn fold_dynamic_standard_basis_tensors(
-    point: &[F],
-    alphas: &[F],
-    alpha_invs: &[F],
-) -> (F, F) {
+pub(crate) fn fold_dynamic_standard_basis_tensors<S: Scalar, I: Into<S> + From<S> + Copy>(
+    point: &[I],
+    alphas: &[I],
+    alpha_invs: &[I],
+) -> (I, I) {
+    let point = point.iter().map(|i| (*i).into()).collect::<Vec<S>>();
+    let alphas = alphas.iter().map(|i| (*i).into()).collect::<Vec<S>>();
+    let alpha_invs = alpha_invs.iter().map(|i| (*i).into()).collect::<Vec<S>>();
     let nu = point.len() / 2 + 1;
     debug_assert_eq!(alphas.len(), nu);
     debug_assert_eq!(alpha_invs.len(), nu);
     let lo_fold = if point.is_empty() {
         alphas[0]
     } else {
-        point.iter().zip(alphas).map(|(v, a)| v + a).product()
+        point.iter().zip(alphas).map(|(v, a)| *v + a).product()
```

Comment:
I think you can remove the clone on line 169 with 
```rust
.zip(alpha_invs.clone())
```

by making this change here:

```rust
point.iter().zip(&alphas).map(|(v, a)| *v + *a).product()
```

Reply:
That's not working for me. Was that working for you locally?